source/common/modules/markdown-editor/renderers/click-and-select.ts

```diff
@@ -31,9 +31,34 @@ export default function clickAndSelect (view: EditorView): (event: MouseEvent) =
       return
     }
 
-    const { top, left, bottom, right } = target.getBoundingClientRect()
-    const fromPos = view.posAtCoords({ x: left, y: top })
-    const toPos = view.posAtCoords({ x: right, y: bottom })
+    // The thing we're clicking on may span multiple lines in the editor; for
+    // example, clicking on a citation that has a (soft) line-wrap in the middle
+    // of the rendered text.
+    //
+    // Grabbing the `getBoundingClientRect` is too coarse in such cases because
+    // the selection will span the *entirety* of both lines rather than just the
+    // text of the rendered citation (in this example).
+    //
+    // So, we need to find the *smallest* rectangle that applies in the list of
+    // all possible rectangles for the target; continuing with the citation
+    // example, clicking on a citation that has a (soft) line-wrap in the middle
+    // of the rendered text *must* only select (highlight) the rectangle of the
+    // citation text itself.
+    const rects = target.getClientRects()
+    let fromPos = null
+    let toPos = null
+
+    for (const rect of rects) {
```

Comment:
Given how multiline spans work … wouldn't it be easier to check for the length of the client rects? And, if it's only one, set from and to as I did before, and, if it's at least two, take the first and last for the coordinates…?

Reply:
I'll check that now and report back today 👍